src/main/java/ch/njol/skript/expressions/ExprScript.java

```diff
@@ -1,93 +1,141 @@
-/**
- *   This file is part of Skript.
- *
- *  Skript is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Skript is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Copyright Peter Güttinger, SkriptLang team and contributors
- */
 package ch.njol.skript.expressions;
 
+import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptCommand;
 import ch.njol.skript.doc.Description;
-import ch.njol.skript.doc.Events;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.parser.ParserInstance;
+import org.jetbrains.annotations.Nullable;
 import org.skriptlang.skript.lang.script.Script;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 import org.bukkit.event.Event;
-import org.jetbrains.annotations.Nullable;
 
-@Name("Script Name")
-@Description("Holds the current script's name (the file name without '.sk').")
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Name("Script")
+@Description({"The current script, or a script from its (file) name.",
+	"If the script is enabled or disabled (or reloaded) this reference will become invalid.",
+	"Therefore, it is recommended to obtain a script reference <em>when needed</em>."})
```

Comment:
I originally had that and removed it because I realised you probably didn't _want_ a reference that acts like that -- it can randomly change from what you expect it to represent, plus nothing can be done about referencing a script that is disabled.
Dead references still work for most stuff (enabling/disabling, etc.) it's just that the data inside isn't indicative of what's actually there anymore.

Reply:
Yeah, it's probably fine as-is